src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```
@@ -0,0 +1,1202 @@
+
+        if (fileNamesToLoad == null) {
+            LOG.error("BatchInputFileService.listInputFileNamesWithDoneFile(" +
+                    batchInputFileType.getFileTypeIdentifer() + ") returned NULL which should never happen.");
+            throw new RuntimeException("BatchInputFileService.listInputFileNamesWithDoneFile(" +
+                    batchInputFileType.getFileTypeIdentifer() + ") returned NULL which should never happen.");
+        }
+
+        for (String inputFileName : fileNamesToLoad) {
+            if (StringUtils.isBlank(inputFileName)) {
+                LOG.error("One of the file names returned as ready to process [" + inputFileName +
+                        "] was blank.  This should not happen, so throwing an error to investigate.");
+                throw new RuntimeException("One of the file names returned as ready to process [" + inputFileName +
+                        "] was blank.  This should not happen, so throwing an error to investigate.");
+            }
+        }
+
+        return fileNamesToLoad;
+    }
+
+    /**
+     * Accepts a file name and returns a byte-array of the file name contents, if possible.
+     *
+     * Throws RuntimeExceptions if FileNotFound or IOExceptions occur.
+     *
+     * @param fileName String containing valid path & filename (relative or absolute) of file to load.
+     * @return A Byte Array of the contents of the file.
+     */
+    protected byte[] safelyLoadFileBytes(String fileName) {
+
+        InputStream fileContents;
+        byte[] fileByteContent;
+        try {
+            fileContents = new FileInputStream(fileName);
+        } catch (FileNotFoundException e1) {
+            LOG.error("Batch file not found [" + fileName + "]. " + e1.getMessage());
+            throw new RuntimeException("Batch File not found [" + fileName + "]. " + e1.getMessage());
+        }
+        try {
+            fileByteContent = IOUtils.toByteArray(fileContents);
```

IOUtils.toByteArray() may not close the stream when done, so I would advise adding a "finally" block to close the stream. I think there are several places in KFS that don't add the explicit close for this type of byte loading. IOUtils does have a closeQuietly() method to help with closing the stream and ignoring exceptions.